### PR TITLE
UnivariateFinite arrays, Take II

### PR DESCRIPTION
**Pull request by ablaom**
To be merged before #46

### CODE REVIEW

File: src/data_utils.jl [hunk 1]
```diff
@@ -332,16 +332,23 @@ UnivariateFinite(::LightInterface, a...; kwargs...) =
...
 const UNIVARIATE_FINITE_VECTOR_DOCSTRING =
 """
-    UnivariateFiniteVector(scores, classes)
+    UnivariateFiniteVector(classes, p; pool=nothing, ordered=false)
 
 Container for UnivariateFinite elements optimised for efficiency.
 Accessing a single element will construct and return the corresponding
 UnivariateFinite lazily.
+
+Here the probabalities `p` should be an array with `size(p, 2) = N`,
```
Comment (tlienart) [inline src/data_utils.jl]: "probabilities" (I would prefer `scores` as the interpretation as a probability is a choice, not a  necessity); and I would  suggest: `size(p, 2) = C` where `C = length(classes)` and rows sum to  one. Btw this is boring but I think we should try to  agree on a set of common symbols (e.g. `C` for number of classes `K` for number of clusters, `n`, `p` for dimensions,...)

**Comment by tlienart** [comment]
Cool, I'll work on the PR to adjust to this :+1:

File: src/data_utils.jl (hunk 1 above)
Comment (ablaom) [reply src/data_utils.jl]: I get your philosophical point. However these are probabilities in the strictest, mathematical sense (they sum to one and hence define a probability measure on the space of classes).  If we say "scores" we the also need to constantly say "and they must sum to one." Also, this is the terminology used throughout Distributions.jl, which we are extending. 

Point taken about the notation. It's a bit of a mess in this particular part of the code. Happy to go with your choice here.

**Comment by ablaom** [comment]
@tlienart Do you want me to finish this PR or leave to you?

**Pull request merged**